django_admin_lightweight_date_hierarchy/admin.py

```diff
@@ -0,0 +1,78 @@
+import re
+import datetime
+
+from django.conf import settings
+from django.contrib import admin
+from django.utils import timezone
+
+
+def get_date_range_for_heirarchy(date_hierarchy, tz):
+    """Generate date range for date heirarchy.
+
+    date_hierarchy <dict>:
+        year (int, required)
+        month (int)
+        day (int)
+    tz <timezone or None>:
+        When none dates remain naive.
```

Comment:
"The timezone in which to generate the datetimes. If None, the datetimes will be naive."